### PR TITLE
ViveController: tracking events

### PR DESCRIPTION
**Pull request by spiderworm**
Added trackingstart andtrackingend events for controller tracking gained/loss. Added isTracking internal state and turn off controller visibility when tracking is lost. Button updates still occur if possible even when tracking is lost.

Disclaimer: I will not be able to actually TEST these changes until I am back at work. So maybe don't merge yet, or verify the changes work as intended first, before merging.

### CODE REVIEW

**Comment by spiderworm** [comment]
Would like to note: with this patch, if the programmer wants to force the controllers to be visible even when tracking is lost, they can still do that. They can achieve this by setting `.visible = true` in their update loop right after calling the controller `.update()`.

File: examples/js/ViveController.js [hunk 1]
```diff
@@ -57,23 +69,47 @@ THREE.ViveController = function ( id ) {
...
 
 		gamepad = findGamepad( id );
 
-		if ( gamepad !== undefined && gamepad.pose !== null ) {
+		var oldTracking = isTracking;
+		checkTracking();
+
+		if (isTracking !== oldTracking) {
+
+			scope.dispatchEvent( { type: 'trackingchanged' } );
```
Comment (mrdoob) [inline examples/js/ViveController.js]: One could also achieve this by doing...

```js
controls.addEventListener( 'trackingstart', onTrackingChanged );
controls.addEventListener( 'trackingend', onTrackingChanged );
```

Comment (spiderworm) [inline examples/js/ViveController.js]: Fixed.

**Comment by mrdoob** [comment]
Closing. Better to focus on WebXR support now.
Thanks though!

**Pull request closed without merging**